lib/vintage_net_mobile/modem/automatic/discovery.ex

```diff
@@ -0,0 +1,58 @@
+defmodule VintageNetMobile.Modem.Automatic.Discovery do
+  @moduledoc false
+
+  # A process for trying to discover a the type of modem and some basic
+  # information to allow dynamic configuration items
+
+  use GenServer
+
+  alias Circuits.UART
+
+  @quectel_vender_id 0x2C7C
+
+  @modems %{
+    {@quectel_vender_id, 0x0125} => VintageNetMobile.Modem.QuectelEC25,
+    {@quectel_vender_id, 0x0296} => VintageNetMobile.Modem.QuectelBG96
+  }
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(args) do
+    modem = detect_modem()
```

Comment:
If there is no modem actually attached and detected, this will run forever and block the `start_link` or `start` callback.

This should prob be done in a `handle_continue` callback and let it block there.

Reply:
Good catch I did not consider that, I will update.

Reply:
In theory, it should require the interface before attempting to start it because of this line:
https://github.com/nerves-networking/vintage_net_mobile/blob/main/lib/vintage_net_mobile.ex#L132

I guess it depends on if `wwan0` will ever come up because of the drivers, or only if the kernel detects a modem as well

Reply:
Yeah, I did some testing and without a modem in VintageNet does not try to start the interface.

However, I can see some unsuspecting/crazy edge case issues here and I think I will make this a little more robust given the risk associated with networking.

Reply:
@jjcarstens I updated to not block the init callback for the discovery server and add a little bit of backoff timing on enumerating the uarts.